wannierberri/__system.py

```diff
@@ -284,23 +289,130 @@ def set_symmetry(self,symmetry_gen=[]):
+                raise ValueError("one should not specify both wannier_centers_cart and wannier_centers_reduced")
+            else:
+                self.wannier_centers_reduced = self.wannier_centers_cart.dot(np.linalg.inv(self.real_lattice))
+        elif self.wannier_centers_reduced is not None:
+                self.wannier_centers_cart = self.wannier_centers_reduced.dot(self.real_lattice)
+        elif hasattr(self,"wannier_centers_cart_auto"):
+                self.wannier_centers_cart = self.wannier_centers_cart_auto
+                self.wannier_centers_reduced = self.wannier_centers_cart.dot(np.linalg.inv(self.real_lattice))
+#        self.wannier_centers_cart*=00.01
+#        self.wannier_centers_reduced*=00.01
+#        print ("Wannier_centers\n",self.wannier_centers_cart,self.wannier_centers_reduced)
+        if self.use_wcc_phase: 
+            if self.wannier_centers_cart is None:
+                raise ValueError("use_wcc_phase = True, but the wannier centers could not be detyermined")
+            if hasattr(self,'AA_R'):
+                AA_R_new = np.copy(self.AA_R)
+                AA_R_new[np.arange(self.num_wann),np.arange(self.num_wann),self.iR0,:] -= self.wannier_centers_cart
+            if hasattr(self,'BB_R'):
+                BB_R_new = self.BB_R.copy() - self.HH_R[:,:,:,None]*self.wannier_centers_cart[None,:,None,:]
+            if hasattr(self,'CC_R'):
+                norm = np.linalg.norm(self.CC_R - self.conj_XX_R(self.CC_R))
+                assert norm<1e-10 , f"norm={norm}"
+                assert hasattr(self,'BB_R') , "if you use CC_R, you need also BB_R"
+                T  =  self.wannier_centers_cart[:,None,None,:,None]*self.BB_R[:,:,:,None,:]
+                CC_R_new  =  self.CC_R.copy() + 1.j*sum(   
+                            s*( -T[:,:,:,a,b]   # -t_i^a * B_{ij}^b(R)
+                                -self.conj_XX_R(T[:,:,:,b,a])    # - B_{ji}^a(-R)^*  * t_j^b 
+                                +self.wannier_centers_cart[:,None,None,a]*self.HH_R[:,:,:,None] * self.wannier_centers_cart[None,:,None,b]  # + t_i^a*H_ij(R)t_j^b
+                            )
+                        for (s,a,b) in [(+1,alpha_A,beta_A) , (-1,beta_A,alpha_A)] )
+                norm = np.linalg.norm(CC_R_new - self.conj_XX_R(CC_R_new))
+                assert norm<1e-10 , f"norm={norm}"
+
+
+            # not sure if the following is correct (Stepan)
+            if hasattr(self,'SA_R'):
+                assert hasattr(self,'SS_R') , "if you use SA_R, you need also SS_R"
+                SA_R_new  =  self.SA_R.copy() - self.SS_R[:,:,:,:,None]*self.wannier_centers_cart[None,:,None,None,:]
+            if hasattr(self,'SHA_R'):
+                assert hasattr(self,'SH_R') , "if you use SA_R, you need also SH_R"
+                SHA_R_new  =  self.SHA_R.copy() - self.SS_R[:,:,:,:,None]*self.wannier_centers_cart[None,:,None,None,:]
+
+            for X in ['AA','BB','CC','SA','SHA']:
```

Comment:
Mainly because to modify CC, we need the old values of BB. But to make sure we do not run into such collisions for other quantities, we generally do modificarions using all XX_R's , and in the end assign XX_R=XX_R_new